apps/mosaico/src/Mosaico/Epaper.purs

```diff
@@ -54,7 +62,7 @@ component = do
           (Aff.launchAff_ <<< (liftEffect <<< setEntitlements <<< Just
                                <<< fromMaybe Set.empty <<< hush <=< User.getUserEntitlements)) tokens
       pure $ pure unit
-    pure $ render onLogin paper (isNothing user) userAuth entitlements
+    pure $ render onLogin paper (isNothing user && loadingIpAuth) ((user *> userAuth) <|> ipUserAuth) (entitlements <|> ipEntitlements)
```

Comment:
I changed it so that if at least one of the entitlements is a Just then it'll use the union of them.

Reply:
Doesn't that now potentially use the wrong token? If the user is logged in, does not have entitlements on their account, but has ip login enabled, then now it's going to try to use the user's token instead of the ip token, which is going to fail.

Reply:
That's why I suggested flipping the <|> -- the users shouldn't be able to login using IP if they don't have entitlements.